src/app.js

```diff
@@ -1,23 +1,154 @@
-// starting point of the application
-
 const express = require("express");
+const { connectDB } = require("./config/database");
+const User = require("./models/user");
+const { validateSignUpData } = require("./utils/validation");
+const bcrypt = require("bcrypt");
+
+const app = express();
+
+app.use(express.json());
+
+// POST api - signup
+app.post("/signup", async (req, res, next) => {
+  try {
+    // validate the data
+    validateSignUpData(req);
+
+    // hash the password
+    const { password } = req.body;
+    const hasedPass = await bcrypt.hash(password, 10);
 
-const app = express(); // instance of this express JS application
+    // Creating the instance of the User model
+    const user = new User({ ...req.body, password: hasedPass });
```

Comment:
bad idea to spread the entire `req.body` since client could send n number of fields are we are only validating a few in our `validateSignUpData`